src/vec.ts

```diff
@@ -153,44 +149,31 @@ export namespace vec {
   }
 
   /**
-   * Inverts a 3x3 matrix, returning null if the determinant is zero (indicating a degenerate
-   * transformation)
+   * Inverts a 3x3 matrix, returning null if the determinant is zero
+   * (indicating a degenerate transformation)
    */
-  export function matrixInvert(matrix: Matrix): Matrix | null {
-    const a = matrix
-    const out: Matrix = new Array(9) as Matrix
+  export function matrixInvert(a: Matrix): Matrix | null {
+    // Calculate the determinant
+    const mDet = det(a)
+    if (!mDet) return null
+
+    const invDet = 1.0 / mDet
 
     const a00 = a[0],
       a01 = a[1],
       a02 = a[2]
     const a10 = a[3],
       a11 = a[4],
       a12 = a[5]
-    const a20 = a[6],
-      a21 = a[7],
-      a22 = a[8]
-
-    const b01 = a22 * a11 - a12 * a21
-    const b11 = -a22 * a10 + a12 * a20
-    const b21 = a21 * a10 - a11 * a20
-
-    // Calculate the determinant
-    let det = a00 * b01 + a01 * b11 + a02 * b21
-
-    if (!det) return null
-    det = 1.0 / det
-
-    out[0] = b01 * det
-    out[1] = (-a22 * a01 + a02 * a21) * det
-    out[2] = (a12 * a01 - a02 * a11) * det
-    out[3] = b11 * det
-    out[4] = (a22 * a00 - a02 * a20) * det
-    out[5] = (-a12 * a00 + a02 * a10) * det
-    out[6] = b21 * det
-    out[7] = (-a21 * a00 + a01 * a20) * det
-    out[8] = (a11 * a00 - a01 * a10) * det
 
-    return out
+    return matrixCreate(
+      invDet * a11,
+      invDet * -a10,
+      invDet * -a01,
+      invDet * a00,
+      invDet * (a12 * a01 - a02 * a11),
```

Comment:
the translation back is more "complicated" because the translation happens along the NEW basis vectors, while the translation to get there happened along the old basis vectors.